tests/Doctrine/Tests/ORM/Functional/Ticket/GH10348Test.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class GH10348Test extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH10348ChildEntity::class,
+            GH10348ParentEntity::class,
+        ]);
+    }
+
+    public function testCanRemoveParentWithChildRelatesToOwnEntity(): void
+    {
+        $child1         = new GH10348ChildEntity();
+        $child2         = new GH10348ChildEntity();
+        $child2->origin = $child1;
+
+        $parent = new GH10348ParentEntity();
+        $parent->addChild($child1)->addChild($child2);
+
+        $this->_em->persist($parent);
+        $this->_em->flush();
+
+        $parent = $this->_em->find(GH10348ParentEntity::class, $parent->id);
+
+        $this->_em->remove($parent);
+
+        $this->expectNotToPerformAssertions();
+        $this->_em->flush();
+    }
+}
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="gh10348_child_entities")
+ */
+class GH10348ChildEntity
```

Comment:
I think the vocabulary parent-child is confusing as there is no PHP inheritance here.

Reply:
I've been thinking about this but was unable to come up with names that would better reveal the intentions. Do you have any suggestions?

https://github.com/doctrine/orm/issues/10348#issuecomment-1455678557 shows the object graph and the problem. Yes, we have a self-referencing association between the "child" classes, but both of them also depend on the "parent" class.

Reply:
Sure!

```suggestion
class GH10348Person
```